ouroboros-consensus/src/Ouroboros/Consensus/Util.hs

```diff
@@ -152,6 +153,11 @@ dropLast n = reverse . drop (fromIntegral n) . reverse
 firstJust :: forall a b f. Foldable f => (a -> Maybe b) -> f a -> Maybe b
 firstJust f = asum . fmap f . toList
 
+allEqual :: Eq a => [a] -> Bool
```

Comment:
Top level recursion prevents this from getting a unfolding by default.

This means it won't be specialized at call sites.

Reply:
We're a very long way away from having to do that level of optimization :)